src/hotspot/share/gc/shenandoah/shenandoahGenerationalHeap.cpp

```diff
@@ -96,3 +116,153 @@ void ShenandoahGenerationalHeap::stop() {
   regulator_thread()->stop();
   ShenandoahHeap::stop();
 }
+
+ShenandoahGenerationalHeap::TransferResult ShenandoahGenerationalHeap::balance_generations() {
```

Comment:
To a related comment on signed value of surplus (or deficit, as you will), one might probably change this if its only current role is to carry the directional information (by compressing all of that information into a sign, with canonical direction based on convention). You could still use this to make logging easier, though.

Reply:
I'll pick up the surplus/deficit change in a subsequent PR.